crowbar_framework/app/models/service_object.rb

```diff
@@ -187,8 +191,8 @@ def elements_not_ready(nodes, pre_cached_nodes = {})
     [ delay, pre_cached_nodes ]
   end
 
-  def add_pending_elements(bc, inst, elements, queue_me, pre_cached_nodes = {})
-    nodes_map = elements_to_nodes_to_roles_map(elements)
+  def add_pending_elements(bc, inst, element_order, elements, queue_me, pre_cached_nodes = {})
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for add_pending_elements is too high. [31.59/25]<br>Metrics/LineLength: Line is too long. [94/80]